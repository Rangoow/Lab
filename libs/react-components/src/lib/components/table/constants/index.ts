export * from './table-demo-data';
