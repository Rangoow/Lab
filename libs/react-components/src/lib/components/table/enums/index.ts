export * from './demo-boolean-tag.enums';
export * from './demo-status-tag.enums';
