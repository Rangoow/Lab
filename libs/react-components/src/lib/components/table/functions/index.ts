export * from './get-table-demo-columns-def';
