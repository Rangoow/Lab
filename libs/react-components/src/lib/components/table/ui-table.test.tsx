import { ColumnDef } from '@tanstack/react-table';
import { RenderResult, render, screen } from '@testing-library/react';
import { UI_TABLE_DEMO_DATA } from './constants';
import { EDemoStatusTag } from './enums';
import { getTableDemoColumnsDef } from './functions';
import { IUiTableDemo } from './interfaces';
import { UiTable } from './ui-table.component';

describe('uiTable', (): void => {
  let tableData: IUiTableDemo[];
  let tableColumns: ColumnDef<IUiTableDemo>[];

  beforeEach((): void => {
    tableData = UI_TABLE_DEMO_DATA;
    tableColumns = getTableDemoColumnsDef();
  });

  describe('snapshot testing', (): void => {
    it(`should render correctly and match the snapshot`, (): void => {
      const result: RenderResult = render(<UiTable data={tableData} columns={tableColumns} />);

      expect(result).toMatchSnapshot();
    });
  });

  describe('component testing', (): void => {
    it(`should render the table with correct number of row and columns`, (): void => {
      render(<UiTable data={tableData} columns={tableColumns} />);

      const tableElement: HTMLElement = screen.getByRole('table');
      const numberOfRow: HTMLElement[] = screen.getAllByRole('row');
      const numberOfColumnHeader: HTMLElement[] = screen.getAllByRole('columnheader');

      expect(tableElement).toBeVisible();
      expect(numberOfRow).toHaveLength(tableData.length + 1);
      expect(numberOfColumnHeader).toHaveLength(7);
    });

    it(`should render the string column`, (): void => {
      render(<UiTable data={tableData} columns={tableColumns} />);

      const stringColumnHeaderElement: HTMLElement = screen.getByRole('columnheader', { name: /string header/i });
      const stringCell1: HTMLElement = screen.getByRole('cell', { name: /string_1/i });
      const stringCell2: HTMLElement = screen.getByRole('cell', { name: /string_2/i });
      const stringCell3: HTMLElement = screen.getByRole('cell', { name: /string_3/i });

      expect(stringColumnHeaderElement).toBeVisible();
      expect(stringCell1).toBeVisible();
      expect(stringCell2).toBeVisible();
      expect(stringCell3).toBeVisible();
    });

    it(`should render the number column`, (): void => {
      render(<UiTable data={tableData} columns={tableColumns} />);

      const numberColumnHeaderElement: HTMLElement = screen.getByRole('columnheader', { name: /number header/i });
      const numberCell1: HTMLElement = screen.getByRole('cell', { name: '1' });
      const numberCell2: HTMLElement = screen.getByRole('cell', { name: '2' });
      const numberCell3: HTMLElement = screen.getByRole('cell', { name: '3' });

      expect(numberColumnHeaderElement).toBeVisible();
      expect(numberCell1).toBeVisible();
      expect(numberCell2).toBeVisible();
      expect(numberCell3).toBeVisible();
    });

    it(`should render the image column`, (): void => {
      render(<UiTable data={tableData} columns={tableColumns} />);

      const imageColumnHeaderElement: HTMLElement = screen.getByRole('columnheader', { name: /image header/i });
      const imageCellElements: HTMLElement[] = screen.getAllByRole('img');

      expect(imageColumnHeaderElement).toBeVisible();
      expect(imageCellElements).toHaveLength(3);
    });

    it(`should render the boolean column`, (): void => {
      render(<UiTable data={tableData} columns={tableColumns} />);

      const imageColumnHeaderElement: HTMLElement = screen.getByRole('columnheader', { name: /boolean header/i });
      const activeTagCellElements: HTMLElement[] = screen.getAllByText('TRUE');
      const inactiveTagCellElements: HTMLElement[] = screen.getAllByText('FALSE');

      expect(imageColumnHeaderElement).toBeVisible();
      expect(activeTagCellElements).toHaveLength(1);
      expect(inactiveTagCellElements).toHaveLength(2);
    });

    it(`should render the status tag column`, (): void => {
      render(<UiTable data={tableData} columns={tableColumns} />);

      const imageColumnHeaderElement: HTMLElement = screen.getByRole('columnheader', { name: /status header/i });
      const status1TagCellElements: HTMLElement[] = screen.getAllByText(EDemoStatusTag.STATUS1);
      const status2TagCellElements: HTMLElement[] = screen.getAllByText(EDemoStatusTag.STATUS2);
      const status3TagCellElements: HTMLElement[] = screen.getAllByText(EDemoStatusTag.STATUS3);

      expect(imageColumnHeaderElement).toBeVisible();
      expect(status1TagCellElements).toHaveLength(1);
      expect(status2TagCellElements).toHaveLength(1);
      expect(status3TagCellElements).toHaveLength(1);
    });

    it(`should render the progressbar  column`, (): void => {
      render(<UiTable data={tableData} columns={tableColumns} />);

      const imageColumnHeaderElement: HTMLElement = screen.getByRole('columnheader', { name: /progressbar header/i });
      const progressbarElements: HTMLElement[] = screen.getAllByRole('progressbar');

      expect(imageColumnHeaderElement).toBeVisible();
      expect(progressbarElements).toHaveLength(3);
    });
  });
});
