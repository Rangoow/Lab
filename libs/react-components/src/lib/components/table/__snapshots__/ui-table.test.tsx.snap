// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UiTable > snapshot testing > should render correctly and match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="ui-table"
      >
        <thead
          class="ui-table__header typo_text-3"
        >
          <tr>
            <th
              class="ui-table__header__cell"
            >
              <button
                class="ui-table__header__cell__sortable"
              >
                String header
                <span
                  class="vtmx-arrow-up-down-fill "
                  style="font-size: 24px;"
                />
              </button>
            </th>
            <th
              class="ui-table__header__cell"
            >
              <button
                class="ui-table__header__cell__sortable"
              >
                Number header
                <span
                  class="vtmx-arrow-up-down-fill "
                  style="font-size: 24px;"
                />
              </button>
            </th>
            <th
              class="ui-table__header__cell"
            >
              <button
                class="ui-table__header__cell__non-sortable"
              >
                Image header
              </button>
            </th>
            <th
              class="ui-table__header__cell"
            >
              <button
                class="ui-table__header__cell__sortable"
              >
                Date header
                <span
                  class="vtmx-arrow-up-down-fill "
                  style="font-size: 24px;"
                />
              </button>
            </th>
            <th
              class="ui-table__header__cell"
            >
              <button
                class="ui-table__header__cell__sortable"
              >
                Boolean header
                <span
                  class="vtmx-arrow-up-down-fill "
                  style="font-size: 24px;"
                />
              </button>
            </th>
            <th
              class="ui-table__header__cell"
            >
              <button
                class="ui-table__header__cell__sortable"
              >
                Status header
                <span
                  class="vtmx-arrow-up-down-fill "
                  style="font-size: 24px;"
                />
              </button>
            </th>
            <th
              class="ui-table__header__cell"
            >
              <button
                class="ui-table__header__cell__sortable"
              >
                ProgressBar header
                <span
                  class="vtmx-arrow-up-down-fill "
                  style="font-size: 24px;"
                />
              </button>
            </th>
          </tr>
        </thead>
        <tbody
          class="ui-table__body"
        >
          <tr
            class="ui-table__body__row"
          >
            <td
              class="ui-table__body__cell typo_text-3"
            >
              string_1
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              1
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <p>
                <img
                  alt="this is an image cell"
                  class="image_cell"
                  src="https://contents.mediadecathlon.com/p2307093/k$95b46fa90b4ee1ffcacc8fdff4371e30/sq/ballon-ligue-1-uber-eats-officiel-club-2022-2023-fifa-basic-hybride-taille-5.jpg?f=3000x3000"
                />
              </p>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <p>
                01/01/2024
              </p>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <span
                class="vtmn-tag vtmn-tag_variant--decorative_brick vtmn-tag_size--medium "
              >
                FALSE
              </span>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <span
                class="vtmn-tag vtmn-tag_variant--decorative_emerald vtmn-tag_size--medium "
              >
                STATUS 1
              </span>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <div
                class="progressbar__container"
              >
                <span
                  class="typo_text-3"
                >
                  50
                   / 
                  100
                </span>
                <progress
                  aria-labelledby="progress-bar-text"
                  max="100"
                  value="50"
                />
              </div>
            </td>
          </tr>
          <tr
            class="ui-table__body__row"
          >
            <td
              class="ui-table__body__cell typo_text-3"
            >
              string_2
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              2
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <p>
                <img
                  alt="this is an image cell"
                  class="image_cell"
                  src="https://contents.mediadecathlon.com/p2561179/k$ecc2eabea2a4941ff4b1bfaeca52b963/sq/velo-ville-electrique-a-moteur-automatique-owuru-cadre-bas-ld-920-e.jpg?f=3000x3000"
                />
              </p>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <p>
                02/01/2024
              </p>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <span
                class="vtmn-tag vtmn-tag_variant--decorative_jade vtmn-tag_size--medium "
              >
                TRUE
              </span>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <span
                class="vtmn-tag vtmn-tag_variant--decorative_gold vtmn-tag_size--medium "
              >
                STATUS 2
              </span>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <div
                class="progressbar__container"
              >
                <span
                  class="typo_text-3"
                >
                  50
                   / 
                  200
                </span>
                <progress
                  aria-labelledby="progress-bar-text"
                  max="200"
                  value="50"
                />
              </div>
            </td>
          </tr>
          <tr
            class="ui-table__body__row"
          >
            <td
              class="ui-table__body__cell typo_text-3"
            >
              string_3
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              3
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <p>
                <img
                  alt="this is an image cell"
                  class="image_cell"
                  src="https://contents.mediadecathlon.com/p2530633/k$449fdab7d9007d9bae37e8ae17ea3cb3/sq/ballon-de-handball-molten-hx3400-ffhb-taille-2.jpg?f=3000x3000"
                />
              </p>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <p>
                03/01/2024
              </p>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <span
                class="vtmn-tag vtmn-tag_variant--decorative_brick vtmn-tag_size--medium "
              >
                FALSE
              </span>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <span
                class="vtmn-tag vtmn-tag_variant--decorative_cobalt vtmn-tag_size--medium "
              >
                STATUS 3
              </span>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <div
                class="progressbar__container"
              >
                <span
                  class="typo_text-3"
                >
                  150
                   / 
                  200
                </span>
                <progress
                  aria-labelledby="progress-bar-text"
                  max="200"
                  value="150"
                />
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="ui-table"
    >
      <thead
        class="ui-table__header typo_text-3"
      >
        <tr>
          <th
            class="ui-table__header__cell"
          >
            <button
              class="ui-table__header__cell__sortable"
            >
              String header
              <span
                class="vtmx-arrow-up-down-fill "
                style="font-size: 24px;"
              />
            </button>
          </th>
          <th
            class="ui-table__header__cell"
          >
            <button
              class="ui-table__header__cell__sortable"
            >
              Number header
              <span
                class="vtmx-arrow-up-down-fill "
                style="font-size: 24px;"
              />
            </button>
          </th>
          <th
            class="ui-table__header__cell"
          >
            <button
              class="ui-table__header__cell__non-sortable"
            >
              Image header
            </button>
          </th>
          <th
            class="ui-table__header__cell"
          >
            <button
              class="ui-table__header__cell__sortable"
            >
              Date header
              <span
                class="vtmx-arrow-up-down-fill "
                style="font-size: 24px;"
              />
            </button>
          </th>
          <th
            class="ui-table__header__cell"
          >
            <button
              class="ui-table__header__cell__sortable"
            >
              Boolean header
              <span
                class="vtmx-arrow-up-down-fill "
                style="font-size: 24px;"
              />
            </button>
          </th>
          <th
            class="ui-table__header__cell"
          >
            <button
              class="ui-table__header__cell__sortable"
            >
              Status header
              <span
                class="vtmx-arrow-up-down-fill "
                style="font-size: 24px;"
              />
            </button>
          </th>
          <th
            class="ui-table__header__cell"
          >
            <button
              class="ui-table__header__cell__sortable"
            >
              ProgressBar header
              <span
                class="vtmx-arrow-up-down-fill "
                style="font-size: 24px;"
              />
            </button>
          </th>
        </tr>
      </thead>
      <tbody
        class="ui-table__body"
      >
        <tr
          class="ui-table__body__row"
        >
          <td
            class="ui-table__body__cell typo_text-3"
          >
            string_1
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            1
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <p>
              <img
                alt="this is an image cell"
                class="image_cell"
                src="https://contents.mediadecathlon.com/p2307093/k$95b46fa90b4ee1ffcacc8fdff4371e30/sq/ballon-ligue-1-uber-eats-officiel-club-2022-2023-fifa-basic-hybride-taille-5.jpg?f=3000x3000"
              />
            </p>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <p>
              01/01/2024
            </p>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <span
              class="vtmn-tag vtmn-tag_variant--decorative_brick vtmn-tag_size--medium "
            >
              FALSE
            </span>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <span
              class="vtmn-tag vtmn-tag_variant--decorative_emerald vtmn-tag_size--medium "
            >
              STATUS 1
            </span>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <div
              class="progressbar__container"
            >
              <span
                class="typo_text-3"
              >
                50
                 / 
                100
              </span>
              <progress
                aria-labelledby="progress-bar-text"
                max="100"
                value="50"
              />
            </div>
          </td>
        </tr>
        <tr
          class="ui-table__body__row"
        >
          <td
            class="ui-table__body__cell typo_text-3"
          >
            string_2
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            2
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <p>
              <img
                alt="this is an image cell"
                class="image_cell"
                src="https://contents.mediadecathlon.com/p2561179/k$ecc2eabea2a4941ff4b1bfaeca52b963/sq/velo-ville-electrique-a-moteur-automatique-owuru-cadre-bas-ld-920-e.jpg?f=3000x3000"
              />
            </p>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <p>
              02/01/2024
            </p>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <span
              class="vtmn-tag vtmn-tag_variant--decorative_jade vtmn-tag_size--medium "
            >
              TRUE
            </span>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <span
              class="vtmn-tag vtmn-tag_variant--decorative_gold vtmn-tag_size--medium "
            >
              STATUS 2
            </span>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <div
              class="progressbar__container"
            >
              <span
                class="typo_text-3"
              >
                50
                 / 
                200
              </span>
              <progress
                aria-labelledby="progress-bar-text"
                max="200"
                value="50"
              />
            </div>
          </td>
        </tr>
        <tr
          class="ui-table__body__row"
        >
          <td
            class="ui-table__body__cell typo_text-3"
          >
            string_3
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            3
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <p>
              <img
                alt="this is an image cell"
                class="image_cell"
                src="https://contents.mediadecathlon.com/p2530633/k$449fdab7d9007d9bae37e8ae17ea3cb3/sq/ballon-de-handball-molten-hx3400-ffhb-taille-2.jpg?f=3000x3000"
              />
            </p>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <p>
              03/01/2024
            </p>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <span
              class="vtmn-tag vtmn-tag_variant--decorative_brick vtmn-tag_size--medium "
            >
              FALSE
            </span>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <span
              class="vtmn-tag vtmn-tag_variant--decorative_cobalt vtmn-tag_size--medium "
            >
              STATUS 3
            </span>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <div
              class="progressbar__container"
            >
              <span
                class="typo_text-3"
              >
                150
                 / 
                200
              </span>
              <progress
                aria-labelledby="progress-bar-text"
                max="200"
                value="150"
              />
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`uiTable > snapshot testing > should render correctly and match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="ui-table"
      >
        <thead
          class="ui-table__header typo_text-3"
        >
          <tr>
            <th
              class="ui-table__header__cell"
            >
              <button
                class="ui-table__header__cell__sortable"
              >
                String header
                <span
                  class="vtmx-arrow-up-down-fill "
                  style="font-size: 24px;"
                />
              </button>
            </th>
            <th
              class="ui-table__header__cell"
            >
              <button
                class="ui-table__header__cell__sortable"
              >
                Number header
                <span
                  class="vtmx-arrow-up-down-fill "
                  style="font-size: 24px;"
                />
              </button>
            </th>
            <th
              class="ui-table__header__cell"
            >
              <button
                class="ui-table__header__cell__non-sortable"
              >
                Image header
              </button>
            </th>
            <th
              class="ui-table__header__cell"
            >
              <button
                class="ui-table__header__cell__sortable"
              >
                Date header
                <span
                  class="vtmx-arrow-up-down-fill "
                  style="font-size: 24px;"
                />
              </button>
            </th>
            <th
              class="ui-table__header__cell"
            >
              <button
                class="ui-table__header__cell__sortable"
              >
                Boolean header
                <span
                  class="vtmx-arrow-up-down-fill "
                  style="font-size: 24px;"
                />
              </button>
            </th>
            <th
              class="ui-table__header__cell"
            >
              <button
                class="ui-table__header__cell__sortable"
              >
                Status header
                <span
                  class="vtmx-arrow-up-down-fill "
                  style="font-size: 24px;"
                />
              </button>
            </th>
            <th
              class="ui-table__header__cell"
            >
              <button
                class="ui-table__header__cell__sortable"
              >
                ProgressBar header
                <span
                  class="vtmx-arrow-up-down-fill "
                  style="font-size: 24px;"
                />
              </button>
            </th>
          </tr>
        </thead>
        <tbody
          class="ui-table__body"
        >
          <tr
            class="ui-table__body__row"
          >
            <td
              class="ui-table__body__cell typo_text-3"
            >
              string_1
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              1
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <p>
                <img
                  alt="this is an image cell"
                  class="image_cell"
                  src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/4.png"
                />
              </p>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <p>
                01/01/2024
              </p>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <span
                class="vtmn-tag vtmn-tag_variant--decorative_brick vtmn-tag_size--medium "
              >
                FALSE
              </span>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <span
                class="vtmn-tag vtmn-tag_variant--decorative_emerald vtmn-tag_size--medium "
              >
                STATUS 1
              </span>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <div
                class="progressbar__container"
              >
                <span
                  class="typo_text-3"
                >
                  50
                   / 
                  100
                </span>
                <progress
                  aria-labelledby="progress-bar-text"
                  max="100"
                  value="50"
                />
              </div>
            </td>
          </tr>
          <tr
            class="ui-table__body__row"
          >
            <td
              class="ui-table__body__cell typo_text-3"
            >
              string_2
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              2
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <p>
                <img
                  alt="this is an image cell"
                  class="image_cell"
                  src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
                />
              </p>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <p>
                02/01/2024
              </p>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <span
                class="vtmn-tag vtmn-tag_variant--decorative_jade vtmn-tag_size--medium "
              >
                TRUE
              </span>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <span
                class="vtmn-tag vtmn-tag_variant--decorative_gold vtmn-tag_size--medium "
              >
                STATUS 2
              </span>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <div
                class="progressbar__container"
              >
                <span
                  class="typo_text-3"
                >
                  50
                   / 
                  200
                </span>
                <progress
                  aria-labelledby="progress-bar-text"
                  max="200"
                  value="50"
                />
              </div>
            </td>
          </tr>
          <tr
            class="ui-table__body__row"
          >
            <td
              class="ui-table__body__cell typo_text-3"
            >
              string_3
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              3
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <p>
                <img
                  alt="this is an image cell"
                  class="image_cell"
                  src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/7.png"
                />
              </p>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <p>
                03/01/2024
              </p>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <span
                class="vtmn-tag vtmn-tag_variant--decorative_brick vtmn-tag_size--medium "
              >
                FALSE
              </span>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <span
                class="vtmn-tag vtmn-tag_variant--decorative_cobalt vtmn-tag_size--medium "
              >
                STATUS 3
              </span>
            </td>
            <td
              class="ui-table__body__cell typo_text-3"
            >
              <div
                class="progressbar__container"
              >
                <span
                  class="typo_text-3"
                >
                  150
                   / 
                  200
                </span>
                <progress
                  aria-labelledby="progress-bar-text"
                  max="200"
                  value="150"
                />
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="ui-table"
    >
      <thead
        class="ui-table__header typo_text-3"
      >
        <tr>
          <th
            class="ui-table__header__cell"
          >
            <button
              class="ui-table__header__cell__sortable"
            >
              String header
              <span
                class="vtmx-arrow-up-down-fill "
                style="font-size: 24px;"
              />
            </button>
          </th>
          <th
            class="ui-table__header__cell"
          >
            <button
              class="ui-table__header__cell__sortable"
            >
              Number header
              <span
                class="vtmx-arrow-up-down-fill "
                style="font-size: 24px;"
              />
            </button>
          </th>
          <th
            class="ui-table__header__cell"
          >
            <button
              class="ui-table__header__cell__non-sortable"
            >
              Image header
            </button>
          </th>
          <th
            class="ui-table__header__cell"
          >
            <button
              class="ui-table__header__cell__sortable"
            >
              Date header
              <span
                class="vtmx-arrow-up-down-fill "
                style="font-size: 24px;"
              />
            </button>
          </th>
          <th
            class="ui-table__header__cell"
          >
            <button
              class="ui-table__header__cell__sortable"
            >
              Boolean header
              <span
                class="vtmx-arrow-up-down-fill "
                style="font-size: 24px;"
              />
            </button>
          </th>
          <th
            class="ui-table__header__cell"
          >
            <button
              class="ui-table__header__cell__sortable"
            >
              Status header
              <span
                class="vtmx-arrow-up-down-fill "
                style="font-size: 24px;"
              />
            </button>
          </th>
          <th
            class="ui-table__header__cell"
          >
            <button
              class="ui-table__header__cell__sortable"
            >
              ProgressBar header
              <span
                class="vtmx-arrow-up-down-fill "
                style="font-size: 24px;"
              />
            </button>
          </th>
        </tr>
      </thead>
      <tbody
        class="ui-table__body"
      >
        <tr
          class="ui-table__body__row"
        >
          <td
            class="ui-table__body__cell typo_text-3"
          >
            string_1
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            1
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <p>
              <img
                alt="this is an image cell"
                class="image_cell"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/4.png"
              />
            </p>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <p>
              01/01/2024
            </p>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <span
              class="vtmn-tag vtmn-tag_variant--decorative_brick vtmn-tag_size--medium "
            >
              FALSE
            </span>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <span
              class="vtmn-tag vtmn-tag_variant--decorative_emerald vtmn-tag_size--medium "
            >
              STATUS 1
            </span>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <div
              class="progressbar__container"
            >
              <span
                class="typo_text-3"
              >
                50
                 / 
                100
              </span>
              <progress
                aria-labelledby="progress-bar-text"
                max="100"
                value="50"
              />
            </div>
          </td>
        </tr>
        <tr
          class="ui-table__body__row"
        >
          <td
            class="ui-table__body__cell typo_text-3"
          >
            string_2
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            2
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <p>
              <img
                alt="this is an image cell"
                class="image_cell"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
              />
            </p>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <p>
              02/01/2024
            </p>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <span
              class="vtmn-tag vtmn-tag_variant--decorative_jade vtmn-tag_size--medium "
            >
              TRUE
            </span>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <span
              class="vtmn-tag vtmn-tag_variant--decorative_gold vtmn-tag_size--medium "
            >
              STATUS 2
            </span>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <div
              class="progressbar__container"
            >
              <span
                class="typo_text-3"
              >
                50
                 / 
                200
              </span>
              <progress
                aria-labelledby="progress-bar-text"
                max="200"
                value="50"
              />
            </div>
          </td>
        </tr>
        <tr
          class="ui-table__body__row"
        >
          <td
            class="ui-table__body__cell typo_text-3"
          >
            string_3
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            3
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <p>
              <img
                alt="this is an image cell"
                class="image_cell"
                src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/7.png"
              />
            </p>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <p>
              03/01/2024
            </p>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <span
              class="vtmn-tag vtmn-tag_variant--decorative_brick vtmn-tag_size--medium "
            >
              FALSE
            </span>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <span
              class="vtmn-tag vtmn-tag_variant--decorative_cobalt vtmn-tag_size--medium "
            >
              STATUS 3
            </span>
          </td>
          <td
            class="ui-table__body__cell typo_text-3"
          >
            <div
              class="progressbar__container"
            >
              <span
                class="typo_text-3"
              >
                150
                 / 
                200
              </span>
              <progress
                aria-labelledby="progress-bar-text"
                max="200"
                value="150"
              />
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
